### ISSUE TITLE
pre-commit hooks are not part of the release thus no immutable tags are available

### ISSUE THREAD

**Issue by AbdelrhmanHamouda** [python-poetry/poetry#5292]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

I see in the documentation the mention and use of pre-commit hooks seems to be supported, however, the hooks don't seem to be part of any release so far. This is making it so one is practically forced to put `master` as the version. This is of course a bad practice not to mention that pre-commit has a warning that always reminds you that it is a bad practice. 

The request is simple: kindly add the pre-commit hooks to the release.

**Comment by finswimmer** [comment]
Hello @AbdelrhmanHamouda ,

`pre-commit` hooks were introduced after the latest available poetry version (1.2.0a2). Once we have released a new version, the repository will receive a new tag and one can use it for the `rev` field in the `.pre-commit-config.yaml`.

Until a new release you can also use a commit hash for the `rev` field, e.g.

```yaml
  - repo: https://github.com/python-poetry/poetry
    rev: b06658f
    hooks:
      - id: poetry-check
```

fin swimmer

**Comment by AbdelrhmanHamouda** [comment]
Thank you for the quick answer. Glad to know that this is coming in the next release as well. 


I was using the `a358be7d4cfff6d0823d4e18f867654724b11596` hash before but we ran into issues where in some cases we would get a error related to an import of PY37. so we dropped the pre-commit hook in favour of a ci/cd check. 

The error in case you are interested:
```shell
Traceback (most recent call last):
File "/Users/MYUSER/.cache/pre-commit/repometfxmh5/py_env-python3/bin/poetry", line 5, in <module>
from poetry.console.application import main
File "/Users/MYUSER/.cache/pre-commit/repometfxmh5/py_env-python3/lib/python3.10/site-packages/poetry/console/application.py", line 19, in <module>
from poetry.core.utils._compat import PY37
ImportError: cannot import name 'PY37' from 'poetry.core.utils._compat' (/Users/MYUSER/.cache/pre-commit/repometfxmh5/py_env-python3/lib/python3.10/site-packages/poetry/core/utils/_compat.py)
```
